### PR TITLE
feat(dropdowns): new light/dark mode colors

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Verified `Combobox` with all combinations of: `startIcon` & `endIcon`, `placeholder`, `inputValue`, `focusInset`, `validation`, `isMultiselectable` with tags & more than `maxTags`, and `isDisabled`.

Verified `Option` with all combinations of: `icon`, `isSelected`, `type`, `isDisabled`, and `Option.Meta`.

Verified `OptGroup` with `label` & `icon`.

Ensured that all `Option` combinations were inherited as expected by `Menu` > `Item`.

## Detail

There are two outstanding questions for design re:
- icon + meta colors for `<Option type="add">`
- icon + meta colors for `<Option isDisabled>`

## Checklist

- [x] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] :arrow_left: ~renders as expected with reversed (RTL) direction~
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [ ] :wheelchair: ~tested for WCAG 2.1 AA accessibility compliance~
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

**Comment by lucijanblagonic** [comment]
> icon + meta colors for `<Option isDisabled>`

image

Use the `foreground/disabled` for all three when it's disabled: `icon`, `label`, `meta`

> icon + meta colors for `<Option type="add">`

I would align add and danger here. Original reasoning was to make `danger` stand out. But after the changes we agreed on the focus state (blue indicator for danger), I think it makes sense to have unified treatment for `meta`.

image

For both Add and Danger, it's the same approach.
* `meta` -> `foreground/subtle`
* `label` + `icon` -> `foreground/primary` and `foreground/danger`, depending on the type

The rest looks great. 🙌

**Comment by jzempel** [comment]
thanks @lucijanblagonic .. the PR has been updated with the finalized color design decisions.

**Review by ze-flo — APPROVED** [review]

**Review by geotrev — APPROVED** [review]
Just the one open question. 🚀

File: packages/dropdowns/src/views/combobox/StyledValue.ts [hunk 1]
```diff
@@ -21,8 +21,8 @@ interface IStyledValueProps extends ThemeProps<DefaultTheme> {
   isPlaceholder?: boolean;
 }
 
-const colorStyles = (props: IStyledValueProps) => {
-  const foregroundColor = props.isPlaceholder && getColorV8('neutralHue', 400, props.theme);
+const colorStyles = ({ theme, isPlaceholder }: IStyledValueProps) => {
+  const foregroundColor = isPlaceholder && getColor({ theme, variable: 'foreground.disabled' });
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/dropdowns/src/views/combobox/StyledValue.ts:25]: `styled-components` does the right thing

**Review by geotrev — APPROVED** [review]

**Review by steue — APPROVED** [review]
thanks for making those menu item changes! combobox and menu focus treatments LGTM!

**Review by lucijanblagonic — COMMENTED** [review]
Everything looks good. 🙌

**Pull request merged**